src/main/java/jakarta/validation/ConstraintValidatorInitializationContext.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Jakarta Validation API
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package jakarta.validation;
+
+import java.time.Clock;
+
+/**
+ * Provides contextual data and operations when initializing a constraint validator.
+ *
+ * @author Marko Bekhta
+ * @since 4.0
+ */
+public interface ConstraintValidatorInitializationContext {
+
+	/**
+	 * Returns the provider for obtaining the current time in the form of a {@link Clock},
+	 * e.g. when validating the {@code Future} and {@code Past} constraints.
+	 *
+	 * @return the provider for obtaining the current time, never {@code null}. If no
+	 * specific provider has been configured during bootstrap, a default implementation using
```

Comment:
You probably need to alter the `Configuration` classes to allow setting the provider? Unless that's already there for other purposes?